src/Microsoft.Extensions.DependencyInjection.Abstractions/IDesignTimeServiceProviderFactory.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// A factory for creating the application's <see cref="IServiceProvider"/> at design time. Implement this interface
+    /// to enable design-time services that need access to the application's services. Design-time services will
+    /// automatically discover implementations of this interface.
+    /// </summary>
+    public interface IDesignTimeServiceProviderFactory
```

Comment:
The scenario that a common non-generic interface would formalize is to delegate the factory somewhere else altogether. For say Prism that'd allow better separation of concerns in PrismApplication on .NET Standard, in which `IServiceProviderFactory<TContainerBuilder>` is only half the solution to e.g. light up Autofac on top of DependencyInjection.Abstractions. Today I'm using [this pattern](http://autofac.readthedocs.io/en/latest/integration/aspnetcore.html) which means so far I had taken dependency on DependencyInjection just to use ServiceCollection. Not sure that kind of scenario warrants defining an interface, but if it exists, I'd prefer to use it.